pubsub/awssnssqs/awssnssqs.go

```diff
@@ -483,27 +525,36 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) (msgs
 		}
 		ms = append(ms, m2)
 	}
-	if len(ms) == 0 {
-		// When we return no messages and no error, the portable type will call
-		// ReceiveBatch again immediately. Sleep for a bit to avoid hammering SQS
-		// with RPCs.
-		time.Sleep(noMessagesPollDuration)
-	}
 	return ms, nil
 }
 
 // SendAcks implements driver.Subscription.SendAcks.
 func (s *subscription) SendAcks(ctx context.Context, ids []driver.AckID) error {
+	g, ctx := errgroup.WithContext(ctx)
+	var ctxErr error
+
+Loop:
 	for _, id := range ids {
-		_, err := s.client.DeleteMessage(&sqs.DeleteMessageInput{
-			QueueUrl:      &s.qURL,
-			ReceiptHandle: id.(*string),
-		})
-		if err != nil {
-			return err
+		select {
+		case <-ctx.Done():
+			ctxErr = ctx.Err()
+			break Loop
+		case s.acksem <- struct{}{}:
+			id := id
+			g.Go(func() error {
+				defer func() { <-s.acksem }()
+				_, err := s.client.DeleteMessage(&sqs.DeleteMessageInput{
```

Comment:
There's no DeleteMessageWithContext? If not, note in a comment.

Reply:
There is; not sure why we weren't using it. Done.